tensorflow_probability/python/distributions/categorical.py

```diff
@@ -333,6 +334,13 @@ def _entropy(self):
         _mul_exp(log_probs, log_probs),
         axis=-1)
 
+  def _mean(self):
+    #probs = self.probs_parameter()
```

Comment:
new impl looks good, thanks! please

1) remove the commented out lines
2) add a test in categorical_test.py. you can look at other categorical tests and maybe normal_test.py for a hint of how these should look. think of edge cases, like some zero prob categories, etc.